crates/bevy_ecs/src/query/fetch.rs

```diff
@@ -327,6 +327,16 @@ pub unsafe trait WorldQuery: for<'w> WorldQueryGats<'w> {
         change_tick: u32,
     ) -> <Self as WorldQueryGats<'w>>::Fetch;
 
+    /// This function is safe to call if `Self:ReadOnlyWorldQuery` holds.
```

Comment:
If this is the case, why not move this to `ReadOnlyWorldQuery`?

Reply:
Because its called for  things that aren't `Self: ReadOnlyWorldQuery`. The safety comment isnt "self is ReadOnlyWorldQuery". The safety comment just always holds if `Self: ReadOnlyWorldQuery` does

Reply:
```suggestion
    /// While this function can be called for any query, it is always safe to call if `Self: ReadOnlyWorldQuery` holds.
```